docs/v1/README.md

```diff
@@ -5,62 +5,67 @@ specification document.
 
 
 Files:
-- `P4Runtime-spec.mdk` is the main file. 
-- assets: Figures
-  - `*.odg` - OfficeLibre source drawing file used to export images. These are
+- `P4Runtime-Spec.adoc` is the main file. 
+- resources: 
+  - figs
+    - `*.odg` - OfficeLibre source drawing file used to export images. These are
     bulk-rendered at build time into .svg and .png images via `soffice`
     command-line (required in build environment)
+  - fonts 
+      - `*.ttf` - Type font source file used to export fonts.
+  - theme: 
+      - `*.yaml` - Describes how PDF P4Runtime specification will be displayed.
+      - `*.css`  - Describes how HTML P4Runtime specification will displayed.
+      - `*.bib`  - Bibliography file that contains a list of bibliographical item, such as articles, books, and theses.
 - `Makefile` builds documentation in the build subdirectory
-- `p4.json` is providing custom syntax highlighting for P4. It is a rip off from
-  the cpp.json provided by Madoko (the "extend" clause does not work, my version
-  of Madoko asks for a "tokenizer" to be defined). Style customization for each
-  token can be done using CSS style attributes (see `token.keyword` in
-  `P4Runtime-spec.mdk`).
+
 
 ## Building
 
-The easiest way to render the Madoko specification documentation is to use the
-`p4lang/p4rt-madoko:latest` Docker` image:
+The easiest way to render the AsciiDoc specification documentation is to use the
+`p4lang/p4rt-asciidoc:latest` Docker` image:
 
-    docker run -v `pwd`/docs/v1:/usr/src/p4-spec p4lang/p4rt-madoko:latest make
+    docker run -v `pwd`/docs/v1:/usr/src/p4-spec p4lang/p4rt-asciidoc:latest make
 
 ### Linux
 ```
-sudo apt-get install nodejs
-sudo npm install madoko -g
-sudo apt-get install libreoffice
-sudo apt-get install texlive-science texlive-xetex
-make [all | html | pdf ]
+rvm install ruby-3.3.1
+rvm use 3.3.1
+gem install asciidoctor
+gem install asciidoctor-pdf
+gem install asciidoctor-bibtex
+gem install asciidoctor-mathematical
+gem install prawn-gmagick
+gem install rouge
+gem install asciidoctor-bibtex
+gem install asciidoctor-lists
+gem install prawn-gmagick
+make 
 ```
-In particular (on Ubuntu 16.04 at least), don't try `sudo apt-get install npm`
-because `npm` is already included and this will yield a bunch of confusing error
-messages from `apt-get`.
-
-`dvipng` is used to render "math" inside BibTex (used for bibliography)
-titles, so you will need to install it as well. On Debian-based Linux, it can be
-done with `sudo apt-get install dvipng`.
+You can use the [local
+installation](https://github.com/p4lang/p4-spec/blob/main/p4-16/spec/install-asciidoctor-linux.sh)
+method.
 
 ### MacOS
 
 We use the [local
-installation](http://research.microsoft.com/en-us/um/people/daan/madoko/doc/reference.html#sec-installation-and-usage)
-method. For Mac OS, I installed node.js using Homebrew and then Madoko using
-npm:
+installation](https://github.com/p4lang/p4-spec/blob/main/p4-16/spec/install-asciidoctor-linux.sh)
```

Comment:
I don't believe this link to the install-asciidoctor-linux.sh is useful to someone who is running macOS.  Best to omit it in this section.

Furthermore, unless you have actually tested the install steps documented here on a macOS system, and successfully generated PDF and HTML there, I would recommend replacing this entire section with text like this:

"We do not yet have instructions for generating PDF and HTML from AsciiDoc source on macOS.  You are welcome to contribute documentation for how to do so if you find instructions that work."

Same comment goes for the Windows section.  I don't want to publish untested steps, especially if they do not work.